x/staking/migrations/v5/store.go

```diff
@@ -0,0 +1,47 @@
+package v5
+
+import (
+	"fmt"
+	"strconv"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/prefix"
+	storetypes "cosmossdk.io/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// MigrateStore performs in-place store migrations from v4 to v5.
+func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey) error {
+	store := ctx.KVStore(storeKey)
+	return migrateHistoricalInfoKeys(store, ctx.Logger())
+}
+
+// migrateHistoricalInfoKeys migrate HistoricalInfo keys to binary format
+func migrateHistoricalInfoKeys(store storetypes.KVStore, logger log.Logger) error {
+	// old key is of format:
+	// prefix (0x50) || heightBytes (string representation of height in 10 base)
+	// new key is of format:
+	// prefix (0x50) || heightBytes (byte array representation using big-endian byte order)
+	oldStore := prefix.NewStore(store, types.HistoricalInfoKey)
+
+	oldStoreIter := oldStore.Iterator(nil, nil)
+	defer sdk.LogDeferred(logger, func() error { return oldStoreIter.Close() })
+
+	for ; oldStoreIter.Valid(); oldStoreIter.Next() {
+		strHeight := oldStoreIter.Key()
+
+		intHeight, err := strconv.ParseInt(string(strHeight), 10, 64)
+		if err != nil {
+			return fmt.Errorf("can't parse height from key %q to int64: %v", strHeight, err)
+		}
+
+		newStoreKey := types.GetHistoricalInfoKey(intHeight)
```

Comment:
Ideally, we define (duplicate) `GetHistoricalInfoKey` in the migration package too since `GetHistoricalInfoKey` can change in the future.

Reply:
Got it, fixed.